core-bundle/src/Security/TwoFactor/BackupCodeManager.php

```diff
@@ -61,7 +82,15 @@ public function generateBackupCodes(User $user): ?array
             $backupCodes[] = $this->generateCode();
         }
 
-        $user->backupCodes = json_encode($backupCodes);
+        $user->backupCodes = json_encode(
+            array_map(
+                static function ($backupCode) {
+                    return password_hash($backupCode, PASSWORD_DEFAULT);
```

Comment:
We could already hash the code in the `generateCode()` method, couldn't we? Then we do not have to traverse the array again.

Reply:
I don’t think so. The un-hashed `$backupCodes` get returned at the end of this method. AFAIK because we have to show the user the un-hashed codes (once).